src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -227,9 +235,57 @@ private SwiftCompileStep makeCompileStep(SourcePathResolver resolver) {
         "-emit-module-path",
         modulePath.toString(),
         "-emit-objc-header-path",
-        headerPath.toString(),
+        headerPath.toString()
+      );
+
+    if (shouldGenerateIndex) {
+      compilerCommand.add(
+        "-num-threads",
+        "12",
+        "-index-store-path",
+        "buck-out"
+      );
+
+      // If there are too many output files, we need to use a output file list instead.
+      // Otherwise, we would get an error about command line too long.
+      boolean use_output_filelist = false;
+      PrintWriter output_filelist = null;
+      String output_filelist_path = "/tmp/" + moduleName + "-output-filelist";
+      if (srcs.size() > MAX_OUTPUT_ARG_COUNT) {
+        try {
+          output_filelist = new PrintWriter(output_filelist_path);
+          compilerCommand.add(
+              "-output-filelist",
+              output_filelist_path
+          );
+          use_output_filelist = true;
+        }
+        catch(FileNotFoundException e) {
+          //  Block of code to handle errors
+          System.out.println(e.getMessage());
+        }
+      }
+
+      // For indexing to work properly, we have to specify multiple .o files for each Swift file
+      // to generate corresponding Unit files. Because Xcode expects an Unit file per Swift file.
+      for (SourcePath sourcePath : srcs) {
+        String source = resolver.getRelativePath(sourcePath).toString();
+        String file = Paths.get(source).getFileName().toString();
+        if (use_output_filelist) {
+          output_filelist.println(outputPath.resolve(file + ".o").toString());
```

Comment:
nit: `outputPath.resolve(file + ".o").toString()` can be out of the if-else block

Reply:
👍